content/knowledge_base/tips/reconfigure_offline_collector.md

```diff
@@ -1,24 +1,109 @@
+configuration embedded - so you can use the collector binary to perform any
+operations that Velociraptor would.
 
-Usually the collector is built using the GUI by selecting the correct artifacts and injecting parameters into the embedded configuration file. But sometimes we might want to slightly modify the embedded configuration, and firing up a GUI to rebuild a new collector from scratch is a bit too much work.
+Usually the collector is built using the GUI by selecting the correct artifacts
+and injecting parameters into the embedded configuration file. But sometimes we
+might want to slightly modify the embedded configuration, and firing up a GUI to
+rebuild a new collector from scratch is a bit too much work.
 
-There is actually an easier way to quickly modify the embedded configuration. 
+Here we describe an easy way to quickly modify the embedded configuration, which
+is suitable for small changes in the embedded configuration. While it is
+recommended that you use the GUI to prepare a completely new collector, for
+small tweaks to an existing offline collector this method may be quicker.
 
-1. First extract the existing embedded config from the collector into a local file:
+## General Method
 
+Assuming you have an existing offline collector named
+`Collector_velociraptor-v0.74.3-windows-amd64.exe`:
+
+1. First extract the existing embedded config from the collector into a local
+   file:
+
+```sh
+Collector_velociraptor-v0.74.3-windows-amd64.exe config show > config.yaml
+```
+
+2. Next, edit the config file - for example, you might want to tweak one or two
+   parameters.
+
+3. Finally, repack the new configuration file into a new collector:
+
+```sh
+Collector_velociraptor-v0.74.3-windows-amd64.exe config repack config.yaml new_collector.exe
+```
+
+You can verify that the new collector has the modified configuration using
+`new_collector.exe config show`.
+
+In the example above the `config repack` command repacked the collector config
+into a copy of the binary which invoked the command. This is the default
+behaviour. If you wish to repack into a different binary then please see the
+next section.
+
+Also note that the commands above are invoked using the offline collector binary
+itself, since this is just a normal Velociraptor binary which happens to have an
+embedded config. This is just for convenience - you could use any Velociraptor
+binary on any platform to do the config extraction or repacking, provided you
+also supply it with the target binary that it will use in generating the output
+file (see next section).
+
+
+### Repacking to a different binary
+
+You can use the `--exe` flag to specify a different target binary. This allows
+you to transfer an existing collector config to a different architecture, and/or
+to a newer binary version. For example:
+
+```sh
+# from Windows amd64 to i386
+velociraptor-v0.74.3-windows-amd64.exe config repack --exe velociraptor-v0.74.3-windows-386.exe config.yaml new_collector.exe
```

Comment:
This is problematic as it won't be able to copy any embedded binaries so it can still fail. I am not sure we want to encourage this procedure. It is always better to build from scratch and it is just as easy

Reply:
All of these use cases are covered by the generic collector which should be our recommended number 1 option in all cases (due to limitations in repacking executables in general).